ipams/windowsipam/windowsipam.go

```diff
@@ -70,8 +71,13 @@ func (a *allocator) RequestAddress(poolID string, prefAddress net.IP, opts map[s
 	if err != nil {
 		return nil, nil, err
 	}
-	if prefAddress == nil {
+
+	// TODO Windows: Remove this once the bug in docker daemon is fixed
+	// that causes it to throw an exception on nil gateway
+	if opts["RequestAddressType"] == netlabel.Gateway {
```

Comment:
Also at line 67 in this file, in the debug log, please remove the extra print of `opts["RequestAddressType"]`.
It is not needed, the key:value elements in `opts` are already being fully printed.